libs/telemetry/src/test/java/org/opensearch/telemetry/tracing/DefaultTracerTests.java

```diff
@@ -265,32 +267,41 @@ public void testSpanAcrossThreads() {
         );
         DefaultTracer defaultTracer = new DefaultTracer(tracingTelemetry, spanTracerStorage);
 
-        executorService.execute(() -> {
+        Future<?> asyncTask = executorService.submit(() -> {
             // create a span
             Span span = defaultTracer.startSpan(new SpanCreationContext("span_name_t_1", Attributes.EMPTY));
             SpanScope spanScope = defaultTracer.withSpanInScope(span);
             spanRef.set(span);
 
-            executorService.execute(() -> {
+            Future<?> asyncTask1 = executorService.submit(() -> {
                 Span spanT2 = defaultTracer.startSpan(new SpanCreationContext("span_name_t_2", Attributes.EMPTY));
                 SpanScope spanScopeT2 = defaultTracer.withSpanInScope(spanT2);
                 spanT2Ref.set(spanT2);
 
                 currentSpanRefThread2.set(defaultTracer.getCurrentSpan().getSpan());
 
-                spanT2.endSpan();
                 spanScopeT2.close();
+                spanT2.endSpan();
                 currentSpanRefAfterEndThread2.set(getCurrentSpanFromContext(defaultTracer));
```

Comment:
Since you are using futures, you could assert directly,  no need to use external context:

```
assertEquals(getCurrentSpanFromContext(defaultTracer), equalTo(null)));
```